04_financial-security/county-debt-coll-shares.qmd

```diff
@@ -0,0 +1,352 @@
+
+
+```{r setup}
+library(tidyverse)
+library(readxl)
+library(glue)
+library(here)
+
+# Set output path
+output_path <- here("04_financial-security/final")
+```
+
+## Helper Function
+
+This function handles the shared transformation logic used across all years.
+
+```{r helper-function}
+#' Process debt collections data
+#' 
+#' Transforms raw debt data into standardized long format with subgroups
+#' 
+#' @param df Raw dataframe from Excel import
+#' @param year Numeric year for the data
+#' @param category Debt category ("overall" or "medical")
+process_debt_data <- function(df, year, category) {
+  df |>
+    # Extract state and county FIPS codes
+    mutate(
+      county_fips = as.character(
+        if ("County FIPS" %in% names(df)) `County FIPS`
+        else if ("CountyFIPS" %in% names(df)) CountyFIPS
+        else if ("GEOID" %in% names(df)) GEOID
+        else stop("No FIPS column found")
+      ),
+      state = str_sub(county_fips, start = 1, end = 2),
+      county = str_sub(county_fips, start = 3, end = 5),
+      # Preserve leading zeros
+      state = str_pad(state, width = 2, pad = "0"),
+      county = str_pad(county, width = 3, pad = "0")
+    ) |>
+   # Rename debt collection share columns (names vary by year and category)
+    (\(data) {
+      if (year == 2022) {
+        rename(data,
+          share_debt_coll_all = `Share with Any Debt in Collections, All`,
+          share_debt_coll_majnonwhite = `Share with Any Debt in Collections, Communities of Color`,
+          share_debt_coll_majwhite = `Share with Any Debt in Collections, Majority White Communities`
+        )
+      } else if (year >= 2023 && category == "overall") {
+        rename(data,
+          share_debt_coll_all = `Share with any debt in collections, All`,
+          share_debt_coll_majnonwhite = `Share with any debt in collections, Comm of color`,
+          share_debt_coll_majwhite = `Share with any debt in collections, White comm`
+        )
+      } else if (year >= 2023 && category == "medical") {
+        rename(data,
+          share_debt_coll_all = `Share with medical debt in collections, All`,
+          share_debt_coll_majnonwhite = `Share with medical debt in collections, Comm of color`,
+          share_debt_coll_majwhite = `Share with medical debt in collections, White comm`
+        )
+      } else {
+        stop(glue("Unsupported year/category combination: {year}/{category}"))
+      }
+    })() |>
+    select(state, county, starts_with("share_debt_coll")) |>
+    filter(county != "OID") |>
+    # Reshape to long format by subgroup
+    pivot_longer(
+      cols = starts_with("share_debt_coll"),
+      names_to = "subgroup",
+      names_prefix = "share_debt_coll",
+      values_to = "share_debt_coll"
+    ) |>
+    # Standardize subgroup labels
+    mutate(
+      subgroup = case_match(
+        subgroup,
+        "_all" ~ "All",
+        "_majnonwhite" ~ "Majority Non-White",
+        "_majwhite" ~ "Majority White"
```

Comment:
We had to write bandaid code to fix this last update. https://github.com/UI-Research/mobility-from-poverty/blob/main/10_construct-database/13_construct_county_race-share-geography.qmd

Can we update this to make sure it aligns with the labels here: https://github.com/UI-Research/mobility-from-poverty/blob/main/data/13_mobility-metrics_county_race-share_longitudinal.csv